internal/api/client_test.go

```diff
@@ -185,7 +185,7 @@ func TestDoWithRetry_RequestError(t *testing.T) {
 
 func TestDoWithRetry_429(t *testing.T) {
 	originalTimeout := retryTimeout
-	retryTimeout = 2 * time.Second
```

Comment:
This fix the flaky test

Reply:
Mind me ask why this test was flaky with retry timeout being 2s?

Reply:
Good question. The timeout is set to 2 second, and the retry interval (`RateLimit-Reset`) is set to 1 second. The test is asserting that the request is made exactly 2 times, but in some situation, it made the request 3 times. (I'm not 100% sure how this happens, maybe because its so quick that the request is made in second 0, 1, and 2).
Reducing the timeout to `1.5 seconds` ensure that the request only made twice.